indexing-service/src/main/java/org/apache/druid/indexing/overlord/config/TaskLockConfig.java

```diff
@@ -42,6 +43,13 @@ public class TaskLockConfig
   @JsonProperty
   private int batchAllocationNumThreads = 5;
 
+  /**
+   * Undocumented parameter, used for embedded tests that create high partition numbers without needing to
+   * individually create each segment. Controls the initial partition number to be used for segment allocation.
+   */
+  @JsonProperty
+  private int initialAllocationPartitionNumber = PartitionIds.ROOT_GEN_START_PARTITION_ID;
```

Comment:
If this config is to be used only for tests, I would advise a slightly different strategy which wouldn't require touching all the production classes.

1. Use the `ClusterTestingModule` extension in the embedded test and enable it on the Overlord via `druid.unsafe.cluster.testing=true`.
2. Create a new class `FaultyIndexerSQLMetadataStorageCoordinator extends IndexerSQLMetadataStorageCoordinator` and override allocation methods.
3. In `ClusterTestingModule`, do `binder.bind(IndexerSQLMetadataStorageCoordinator.class).to(FaultyIndexer...);`
4. Pass a test config into the faulty implementation to control the `startPartitionId`.

Reply:
I just pushed a different approach:

1. First insert the test data into a single segment using SQL, with partition 0.
2. Direct insert a new segment into metadata storage that is the same as that first one, but with a partition number of Short.MAX_VALUE
3. Run SQL to insert 3 more segments.
4. Verify all are loaded and queryable.

Reply:
I removed this new parameter I had added.